kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```diff
@@ -386,6 +386,22 @@ func (c *loadBalancerController) syncIPAM() {
 // - Updates the IP addresses in the Service Status to match the IPAM DB.
 func (c *loadBalancerController) syncService(svcKey serviceKey) error {
 	if len(c.ipPools) == 0 {
+		if _, ok := c.allocationTracker.ipsByService[svcKey]; ok {
+			// Last LoadBalancer IPPool was deleted, and we have previously assigned IPs to this service. We need to release the IPs now and update the service status
+			err := c.releaseIPsByHandle(svcKey)
+			if err != nil {
+				return err
+			}
+			svc, err := c.serviceLister.Services(svcKey.namespace).Get(svcKey.name)
```

Comment:
Any chance of the Service not existing? probably want to handle that gracefully here